src/envoy/mixer/grpc_transport.cc

```diff
@@ -51,19 +51,32 @@ inline void CopyHeaderEntry(const HeaderEntry* entry,
   }
 }
 
+// GrpcTransport objects need to be freed in a delayed fashion.
+// This list stores objects pending to be freed.
+thread_local std::vector<GrpcTransport*> pending_freelist;
+
+void FreePendingGrpcObject() {
+  for (GrpcTransport* obj : pending_freelist) {
+    delete obj;
+  }
+  pending_freelist.clear();
+}
+
 }  // namespace
 
 GrpcTransport::GrpcTransport(Upstream::ClusterManager& cm,
                              const HeaderMap* headers)
-    : channel_(NewChannel(cm)), stub_(channel_.get()), headers_(headers) {}
+    : channel_(NewChannel(cm)), stub_(channel_.get()), headers_(headers) {
+  FreePendingGrpcObject();
```

Comment:
This is not so clean. Is it possible to launch a timer to periodically cleanup the free list? This is just leaking memory, because if I do 100 GRPC calls followed by 1hr of silence, I am going to have 100 GrpcTransport objects lying around. I need to do 100 more calls to delete the previous 100 objects